fakeserver/server.go

```diff
@@ -24,20 +22,15 @@ type server struct {
 	router *mux.Router
 }
 
-// Start the server
-func Start(listenOn string) {
-	var wait time.Duration
-	flag.DurationVar(&wait, "graceful-timeout", time.Second*15, "the duration for which the server gracefully wait for existing connections to finish - e.g. 15s or 1m")
-	flag.Parse()
-
-	r := mux.NewRouter()
-
-	s := &server{router: r}
-	s.routes()
+func loggingMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		log.Println(r.RemoteAddr, r.Method, r.RequestURI)
```

Comment:
Does this auto-flush by default? 12-factor also suggests you shouldn't buffer stdout log output.

Reply:
yep, logging to stdout is unbuffered by default